modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -421,7 +420,35 @@ void cv::gimpl::GExecutor::reshape(const GMetaArgs& inMetas, const GCompileArgs&
     ade::passes::PassContext ctx{g};
     passes::initMeta(ctx, inMetas);
     passes::inferMeta(ctx, true);
-    m_ops[0].isl_exec->reshape(g, args);
+
+    // NB: Before reshape islands need to re-init resources for them.
+    auto sorted = m_gim.metadata().get<ade::passes::TopologicalSortData>();
+    for (auto nh : sorted.nodes())
+    {
+        switch (m_gim.metadata(nh).get<NodeKind>().k)
+        {
+        case NodeKind::ISLAND:
+            // NB: Do nothing.
+            break;
+
+        case NodeKind::SLOT:
+            {
+                const auto orig_data_nh
+                    = m_gim.metadata(nh).get<DataSlot>().original_data_node;
+                initResource(nh, orig_data_nh);
+            }
+        break;
+
+        default:
+            GAPI_Assert(false);
+            break;
+        } // switch(kind)
+    } // for(gim nodes)
```

Comment:
Is this a copy-paste of the above constructor code?

As you can see, previously this method has been composed of some smaller methods and was quite short.
Now you've jumped to one or two levels below of its past abstraction level, also duplicated code from ctor.

If this can be isolated and unified - please do so - I believe this reshape method may stay compact and concise as it used to be.

Reply:
In fact it isn't a full copy-past from ctor, there is some difference.
I can propose to wrap it in function.

Reply:
Oh, actually just going through "m_slots" and call `initResource` is enough